webapp/login.go

```diff
@@ -0,0 +1,255 @@
+package webapp
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/gob"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-github/github"
```

Comment:
```suggestion
	"github.com/google/go-github/github/v28"
```

This is how we use a versioned module in Go. Without specifying the version explicitly, you'd get the `+incompatible` marker in `go.mod`.